### PR TITLE
Fix multiple response.WriteHeader calls error in remote read adapter

### PR DESCRIPTION
**Pull request by qingwave**
1. **Error in remote read adapter**
Error in `/documentation/examples/remote_storage/remote_storage_adapter/main.go`. When [w.write](https://github.com/prometheus/prometheus/blob/a2ef8cf2f509b58d140acf9e37f1e906b28582cd/documentation/examples/remote_storage/remote_storage_adapter/main.go#L298) return err, log shows `http: multiple response.WriteHeader calls`. `w.write` and `http.Error(w, err.Error(), http.StatusInternalServerError)` will write header twice，triggers the error.

2. **Bug fix**
Replace the  `http.Error(w, err.Error(), http.StatusInternalServerError)` with log.

### CODE REVIEW

File: documentation/examples/remote_storage/remote_storage_adapter/main.go [hunk 1]
```diff
@@ -296,7 +296,7 @@ func serve(logger log.Logger, addr string, writers []writer, readers []reader) e
 
 		compressed = snappy.Encode(nil, data)
 		if _, err := w.Write(compressed); err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			level.Warn(logger).Log("msg", "Resonse write error", "storage", reader.Name(), "err", err)
```
Comment (simonpasquier) [inline documentation/examples/remote_storage/remote_storage_adapter/main.go]: It would be a better fix to move L294-295 after this block.

Comment (qingwave) [reply documentation/examples/remote_storage/remote_storage_adapter/main.go]: The error is not isolated to L294-295.
 Function [w.Write](https://github.com/golang/go/blob/889aa5eb98ff2a5134f5b075525479da399e5c4f/src/net/http/server.go#L367) will WriteHeader and function [http.Error](https://github.com/golang/go/blob/889aa5eb98ff2a5134f5b075525479da399e5c4f/src/net/http/server.go#L2004) will WriteHeader too.
Function Write
```golang
func (cw *chunkWriter) Write(p []byte) (n int, err error) {
	if !cw.wroteHeader {
		cw.writeHeader(p)
	}
	...
}
```
Function Error
```golang
func Error(w ResponseWriter, error string, code int) {
	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
	w.Header().Set("X-Content-Type-Options", "nosniff")
	w.WriteHeader(code)
	fmt.Fprintln(w, error)
}
```

Comment (simonpasquier) [reply documentation/examples/remote_storage/remote_storage_adapter/main.go]: My bad, I overlooked the code. 

```go
level.Warn(logger).Log("msg", "Error writing response", "storage", reader.Name(), "err", err)
```

And you can remove `return` at L300.

Comment (qingwave) [reply documentation/examples/remote_storage/remote_storage_adapter/main.go]: Okay, I have removed the useless return as your  suggest.

Comment (simonpasquier) [reply documentation/examples/remote_storage/remote_storage_adapter/main.go]: Thanks! Can you also change `"Resonse write error"` to `"Error writing response"` please?

Comment (qingwave) [reply documentation/examples/remote_storage/remote_storage_adapter/main.go]: My fault! Changed.

**Pull request merged**